src/main.rs

```diff
@@ -85,14 +84,22 @@ fn parse_args() -> (String, Box<dyn Client + 'static + Send>) {
     }
     let client_type = &args[1];
     let path = &args[2];
+    let can_interface = &args[3];
```

Comment:
You need to validate that you have three args or this will fatal error

Reply:
Alright, should be fixed.  I added a part that warns before exiting that not enough arguments are specified.